### PR TITLE
New tests to check pm counters are accessible and reporting

### PR DESCRIPTION
**Pull request by JPRichings**
Test for pm counters added:

- Node Energy
- Node Power
- CPU Energy
- CPU Power
- Memory Energy
- Memory Power
- CPU0 temperature
- CPU1 temperature

### CODE REVIEW

File: tests/env/counters_check.py [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+import reframe.utility.sanity as sn
+
+
+@rfm.simple_test
+class CrayCountersEnergyTest(rfm.RunOnlyRegressionTest):
+    """Checks that the Node Energy counter is reporting"""
+
+    descr = "Checks whether the node energy pm counter is accessible and reporting"
+    valid_systems = ["archer2:compute"]
+    valid_prog_environs = ["PrgEnv-cray", "PrgEnv-gnu", "PrgEnv-aocc"]
```
Comment (aturner-epcc) [inline tests/env/counters_check.py:19]: I think you should only choose a single PrgEnv otherwise each test will be run three times in an identical fashion

**Review by aturner-epcc — CHANGES_REQUESTED** [review]
See note on reducing the number of PrgEnv for each test

File: tests/env/counters_check.py (hunk 1 above)
Comment (RuiApostolo) [reply tests/env/counters_check.py]: Does it need `PrgEnv-cray` or could it be `Default`?
https://github.com/EPCCed/epcc-reframe/blob/518b1227148f8c47eebe8d7245fb05443a4831bd/configuration/archer2.py#L147C21-L147C22

Comment (RuiApostolo) [reply tests/env/counters_check.py]: Ignore me, the `Default` environment is only valid for the `login` partition.

Comment (JPRichings) [reply tests/env/counters_check.py]: @aturner-epcc happy for this to be resolved?

**Review by aturner-epcc — APPROVED** [review]
Thanks for the changes, merging....

**Pull request merged**